pkg/tracing/context.go

```diff
@@ -0,0 +1,32 @@
+package tracing
```

Comment:
Can you add a `doc.go` file to this package to explain the behavior? I think I've figured it out, but it's a bit confusing and I don't see very many doc comments.

If I'm correctly, you create a `Trace` at the beginning of the thing you want to trace. You then use the `Span` to create child spans. All spans are managed by the `Trace` and each `Span` only keeps track of its own level. It does not keep track of anything else.

Reply:
Done